docs/tutorial-run-solver.md

```diff
@@ -0,0 +1,180 @@
+5. [Run the Solver](#run-the-solver)
+6. [Advanced Configuration](#advanced-configuration)
+   1. [Orderbook Filter](#Configuring-an-Orderbook-Filter)
+
+---
+
+## Installing Driver Tools
+
+Clone the dex-services repository, and acquire the project requirements
+
+```sh
+git clone https://github.com/gnosis/dex-services.git
+```
+
+We demonstrate how to run the open solver.
+Once comfortable, the open solver can be adapted to suit your own needs from the [open solver repo](https://github.com/gnosis/dex-open-solver).
+
+## Building The Driver
+
+Build the driver (with the open solver dockerhub image) from within the root directory of `dex-services` repo. This command builds the driver while simultanesouly baking the open solver into the the build.
+
+```sh
+docker-compose build --build-arg SOLVER_BASE=gnosispm/dex-open-solver:master stablex-debug
+```
+
+Be aware, your Docker installation may require elevated privileges!
+
+Note that, if `SOLVER_BASE` is not specified, the driver's own internal `NaiveSolver` will be used.
+Alternatively, if you have your own personalized solver, this image would need to be specified during the build.
+
+## Configuring Environment Variables
+
+The following three environment variables are _required_ when running the dex-services project.
+Note that the account corresponding to the private key should be funded with sufficient ETH to pay gas for solution submission.
+
+```sh
+export INFURA_KEY=<your infura key>
+export PRIVATE_KEY=<your private key>
+export NODE_URL=https://mainnet.infura.io/v3/${INFURA_KEY}
+```
+
+By placing these values into an `.env` file and sourcing this file, you can avoid providing them as runtime arguments on every execution.
+
+## Acquiring the Orderbook
+
+It can take a very long time for the driver to load the orderbook and this process is also restricted by infura rate-limits so it is best to supply an existing orderbook file, available for download at
+
+- [mainnet](https://gnosis-dfusion-volume-mainnet.s3.amazonaws.com/stablex_orderbook_mainnet.bin)
+- [rinkeby](https://gnosis-dfusion-volume-rinkeby.s3.amazonaws.com/stablex_orderbook_rinkeby.bin)
+
+Copy this orderbook file into `dex-services/target` and append `ORDERBOOK_FILE=/app/dex-services/target/stablex_orderbook_mainnet.bin` to your `.env` file from the previous section.
+Technically the orderbook file can be saved anywhere, but we have chosen `target` here since it is flagged as an untracked directory in the project's `.gitignore`.
+
+## Run the Solver
+
+First run and enter the solver container
+
+```sh
+docker-compose run -v $PWD/:/app/dex-services stablex-debug
+```
+
+Since the project was mounted inside the container you can make changes to the driver code and they will be directly reflected on every restart.
+
+The driver can now be run via
+
+```sh
+cargo run --bin driver -- --solver-type OpenSolver --node-url $NODE_URL --private-key $PRIVATE_KEY --orderbook-file $ORDERBOOK_FILE
```

Comment:
it might be worth stating that this only works from within the docker container, and how to get there (unless you have the open solver installed on your host system).